tests/components/samsungtv/test_config_flow.py

```diff
@@ -0,0 +1,388 @@
+    assert result["data"][CONF_MANUFACTURER] == "fake2_manufacturer"
+    assert result["data"][CONF_MODEL] == "fake2_model"
+    assert result["data"][CONF_ID] == "fake2_uuid"
+
+
+async def test_ssdp_missing_auth(hass):
+    """Test starting a flow from discovery with authentication."""
+    with patch(
+        "homeassistant.components.samsungtv.config_flow.Remote",
+        side_effect=AccessDenied("Boom"),
+    ), patch("homeassistant.components.samsungtv.config_flow.socket"):
+
+        # confirm to add the entry
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": "ssdp"}, data=MOCK_SSDP_DATA
+        )
+        assert result["type"] == "form"
+        assert result["step_id"] == "confirm"
+
+        # missing authentication
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input="whatever"
+        )
+        assert result["type"] == "abort"
+        assert result["reason"] == "auth_missing"
+
+
+async def test_ssdp_not_supported(hass):
+    """Test starting a flow from discovery for not supported device."""
+    with patch(
+        "homeassistant.components.samsungtv.config_flow.Remote",
+        side_effect=UnhandledResponse("Boom"),
+    ), patch("homeassistant.components.samsungtv.config_flow.socket"):
+
+        # confirm to add the entry
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": "ssdp"}, data=MOCK_SSDP_DATA
+        )
+        assert result["type"] == "form"
+        assert result["step_id"] == "confirm"
+
+        # device not supported
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input="whatever"
+        )
+        assert result["type"] == "abort"
+        assert result["reason"] == "not_supported"
+
+
+async def test_ssdp_not_found(hass):
+    """Test starting a flow from discovery but no device found."""
+    with patch(
+        "homeassistant.components.samsungtv.config_flow.Remote",
+        side_effect=OSError("Boom"),
+    ), patch("homeassistant.components.samsungtv.config_flow.socket"):
+
+        # confirm to add the entry
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": "ssdp"}, data=MOCK_SSDP_DATA
+        )
+        assert result["type"] == "form"
+        assert result["step_id"] == "confirm"
+
+        # device not found
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input="whatever"
+        )
+        assert result["type"] == "abort"
+        assert result["reason"] == "not_found"
+
+
+async def test_ssdp_already_in_progress(hass, remote):
+    """Test starting a flow from discovery twice."""
+
+    # confirm to add the entry
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": "ssdp"}, data=MOCK_SSDP_DATA
+    )
+    assert result["type"] == "form"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=4795320)